src/test/java/BaseTest.java

```diff
@@ -1,18 +1,90 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
 import org.openqa.selenium.firefox.FirefoxDriver;
-import org.openqa.selenium.safari.SafariDriver;
-import org.testng.annotations.BeforeMethod;
-import org.testng.annotations.BeforeSuite;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.*;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.time.Duration;
 
 
 public class BaseTest {
+    public static WebDriver driver = null;
+    public static WebDriverWait wait = null;
+    public static String url = null;
 
     @BeforeSuite
     static void setupClass() {
-        WebDriverManager.chromedriver().setup();
+        //WebDriverManager.chromedriver().setup();
     }
 
-}
+    @BeforeMethod
+    @Parameters({"BaseUrl"})
+    public void launchBrowser(String BaseUrl) throws MalformedURLException {
+        url = BaseUrl;
+        driver = pickBrowser(System.getProperty("browser"));
```

Comment:
I would do it in another way, but your solution is also fine

`driver = pickBrowser("browser");`